### DISCUSSION TITLE
Re-posting problem with simulated weight distribution

### DISCUSSION THREAD

**Discussion by livPKPD** [LAPKB/Pmetrics#89] [General]
Hi, as discussed I am re-posting this so that it is picked up when you get a chance!
I have an oral drug dosed in mg/kg. I do not have weight in my model as it was not significantly associated with the PK in regression analysis. But I want to sim the weights in the population and dose the drug according to the simulated weight distribution.

Model is here:
mod <- PM_model$new(list(
pri = list(
Ka = range(0.1,7),
V = range(0.1,40),
Kcp = range(0.01,20),
Kpc = range(0.01,20),
Kcs = range(0.01,20),
Ksc = range(0.01,20),
Vcns = range(0.1,60),
SCL = range(0.01,10)
),
fa = "FA(1)=WT",
dif = list(
xp1 = "-Ka * X(1)",
xp2 = "- (Kcp + Kcs + (SCL/V))X(2) + (KscX(3)) + (KpcX(4)) + (KaX(1))",
xp3 = "KcsX(2) - KscX(3)",
xp4 = "KcpX(2) - KpcX(4)"
),
cov = "WT",
out = list(
Y1 = list(
value = "X(2)/V",
err = list(
model = additive(3),
assay = c(0.025, 0.1, 0, 0)
)
),
Y2 = list(
value = "X(3)/Vcns",
err = list (
model = additive(3),
assay = c(0.025, 0.1, 0, 0)
)
)
)
))

This is the sim code:

simrunwt<- PM_load(64)

covList = list(
cov = simrunwt$cov$data,
mean = list(wt = 50),
sd = list(wt = 8.24),
limits = list(wt = c(37, 83))
)

setwd(paste(wd, "/Sims", sep = ""))
sim1 <- PM_sim$run(simrunwt$final
, data = "simmgkg.csv"
, model = "genmodel.txt"
, covariate = covList
, limits = NA, nsim = 1000
, predInt= 1
)

And I have also tried this way:

sim1<- simrunwt$sim(covariate = covList
, limits = NA, nsim = 1000
, predInt= 1
)

I have attached my sim datafile "simmgkg.csv" and model file "genmodel.txt" to the email I sent to @Siel (I have tried the S3 method too).

I have also tried adding FA1 as a primary variable in addition to having it in the #FA block.

I have tried adding an additional 'dummy' covariate.

In any of the iterations I have tried, I get the same error code:

DATA STANDARDIZATION REPORT: Data are in full format already.
DATA VALIDATION REPORT:No data errors found.
FIX DATA REPORT:There were no errors to fix in you data file.
Error in match.names(clabs, names(xi)) : names do not match previous names

Do you have any ideas please?! I attended the recent PM course and have the example.R file. When I run this, the sims work perfectly, including with covariates.

**Comment by Siel** [comment]
Fixed the prublem and pushed it to a new branch to further testing.
